### DISCUSSION TITLE
MasterDetailRelation Save Error

### DISCUSSION THREAD

**Discussion by mrokicki-dev** [serenity-is/Serenity#5604] [Help]
I followed the MasterDetail examples in the documentation and Northwind DB, but the record is looking for the parent ID and will not save without it.

I tried overriding the SetInternalFields on the WorkFlowRepository and manually setting the WorkFlowId for each item in the Milestones list. This gets past the validation error, but deletes all the child records every time a new record is updating.

![SerenityMasterDetailError](https://user-images.githubusercontent.com/77282876/113015639-2d3d1880-9197-11eb-9857-be7f5300aa1a.PNG)

**Comment by erehtolleh** [comment]
Have you searched the discussions or issues?  Try this #5390 .

**Reply by mrokicki-dev** [reply]
Yes, I've searched here as well as stack overflow. I have the files mentioned in issues #5390. I've also tried to implement work arounds others have presented to the same issue, but have not been successful.

**Reply by erehtolleh** [reply]
You have childeditor and childeditordialog?  Make sure it opens childeditordialog...

Without additional info, kind of hard to help you any further, good luck.

It should work if you set it up exactly per manual.

**Reply by erehtolleh** [reply]
Also, did you make the ID Hidden?  I noticed if you have it on the form, it expects you to fill it in?  I just created a MasterDetailRelation and it works fine.  You should post your MasterDetailRelation codes and all related files...

**Reply by erehtolleh** [reply]
I think you put the wrong id in your MasterDetailRelation
MasterDetailRelation(foreignKey: "thisi is id of parent in child table", IncludeColumns = "")
